### ISSUE TITLE
Segfault if a function definition has a css comment

### ISSUE THREAD

**Issue by soulan** [sass/libsass#646]
@function text-contrast($bg-color, $dark-text: $dark-text-color, $light-text: $light-text-color) {
    /*
	$bg-brightness: round(red($bg-color) * 0.3) + (green($bg-color) * 0.59) + (blue($bg-color) * 0.11);
	@if abs($bg-brightness) < 200 {
		@return $light-text;
	} @else {
		@return $dark-text;
	}
	*/
	
	@if use-dark-text($bg-color){
		@return $dark-text;
	} @else {
		@return $light-text;
	}

	/*
	$color-brightness: round((red($bg-color) * 300) + (green($bg-color) * 590) + (blue($bg-color) * 110) / 1000);
  	$light-text-brightness: round((red($light-text) * 300) + (green($light-text) * 590) + (blue($light-text) * 110) / 1000);
  	$dark-text-brightness: round((red($dark-text) * 300) + (green($dark-text) * 590) + (blue($dark-text) * 110) / 1000);
  	
  	@if abs($color-brightness - $light-text-brightness) > abs($color-brightness - $dark-text-brightness) {
		@return $light-text;
	} @else {
		@return $dark-text;
	}
	*/
    }

    @mixin contrasted-important($background-color, $dark-text: $dark-text-color, $light-text: $light-text-color){
	background-color: $background-color !important;
	color: text-contrast($background-color, $dark-text, $light-text) !important;
    }

I've got these 2 mixins, when I call contrasted-important in this setup I get a segemntation fault

    @mixin contrasted-important($background-color, $dark-text: $dark-text-color, $light-text: $light-text-color){
	background-color: $background-color !important;
	color: text-contrast($background-color, $dark-text, $light-text);
    }

and in this setup the value of color gets

    color: /*
	$bg-brightness: round(red($bg-color) * 0.3) + (green($bg-color) * 0.59) + (blue($bg-color) * 0.11);
	@if abs($bg-brightness) < 200 {
		@return $light-text;
	} @else {
		@return $dark-text;
	}
	*/;

**Comment by xzyfer** [comment]
Reduced test case

```scss
@function foo() {
  /*
  $bar: 1;
  */
 @return true;
}

a {
  a: foo();
}
```

This cases Libsass to hang indefinitely.

**Comment by xzyfer** [comment]
Spec added https://github.com/sass/sass-spec/pull/241.

**Comment by xzyfer** [comment]
This is fixed and will be in 3.2.
